pkgs/development/python-modules/termplotlib/default.nix

```diff
@@ -19,13 +19,26 @@ buildPythonPackage rec {
   };
 
   format = "pyproject";
-  checkInputs = [ pytestCheckHook numpy exdown gnuplot ];
+  checkInputs = [
+    pytestCheckHook
+    exdown
+    gnuplot
+    numpy
+  ];
   pythonImportsCheck = [ "termplotlib" ];
 
-  # there seems to be a newline in the very front of the output
-  # which causes the test to fail, since it apparently doesn't
-  # strip whitespace. might be a gnuplot choice? sigh...
-  disabledTests = [ "test_plot_lim" ];
+  propagatedBuildInputs = [
+    gnuplot
```

Comment:
Including this in `propagatedBuildInputs` is probably not as robust as it could be. Arguably it's better than it was. There's no guarantee what the `$PATH` in the finally executed python process is going to be and there's always a possibility injecting `gnuplot` into the broader environment causes conflicts.

A more robust way is to substitute the full path of our `gnuplot` binary in to the source wherever it uses `subprocess` to make a call to it, e.g. https://github.com/nschloe/termplotlib/blob/7a440c76eeb8361cb5793e0a6b070c3d528fcd66/src/termplotlib/plot.py#L20

Reply:
I see. Yeah, I was thinking about substituting the path.

I have to keep `gnuplot` in `propagatedBuildInputs`, right?

Reply:
I create a PR which patches the not absolute paths in gnuplot. Hold on.

Reply:
https://github.com/NixOS/nixpkgs/pull/146708

Reply:
This is likely due to `strictDeps = true;`, `propagatedNativeBuildInputs` should propagate the path.

Reply:
If you substitute in the absolute path you can remove it from all inputs.

Reply:
(probably, unless gnuplot has some extra quirk i don't know about like some companion executable it expects in its own path)

Reply:
I went with the patch approach.